docs/cli.rst

```diff
@@ -0,0 +1,287 @@
+
+Schema Handling
+---------------
+
+A schema file can be tested for validity against the schema specification with the ``validate-schema`` command.
+
+.. command-output:: anonlink validate-schema --help
+
+Example
+~~~~~~~
+
+::
+
+     $ anonlink validate-schema clkhash/data/randomnames-schema.json
+     schema is valid
+
+
+Schema files of older versions can be converted to the latest version with the ``convert-schema`` command.
+
+.. command-output:: anonlink convert-schema --help
+
+
+.. _data-generation:
+
+Data Generation
+---------------
+
+The command line tool has a ``generate`` command for generating fake pii data.
+
+.. command-output:: anonlink generate --help
+
+
+::
+
+    $ anonlink generate 1000 fake-pii-out.csv
+    $ head -n 4  fake-pii-out.csv
+    INDEX,NAME freetext,DOB YYYY/MM/DD,GENDER M or F
+    0,Libby Slemmer,1933/09/13,F
+    1,Garold Staten,1928/11/23,M
+    2,Yaritza Edman,1972/11/30,F
+
+
+
+
+A corresponding hashing schema can be generated as well::
+
+    $ anonlink generate-default-schema schema.json
+    $ cat schema.json
+    {
+      "version": 1,
+      "clkConfig": {
+        "l": 1024,
+        "k": 30,
+        "hash": {
+          "type": "doubleHash"
+        },
+        "kdf": {
+          "type": "HKDF",
+          "hash": "SHA256",
+          "salt": "SCbL2zHNnmsckfzchsNkZY9XoHk96P/G5nUBrM7ybymlEFsMV6PAeDZCNp3rfNUPCtLDMOGQHG4pCQpfhiHCyA==",
+          "info": "c2NoZW1hX2V4YW1wbGU=",
+          "keySize": 64
+        }
+      },
+      "features": [
+        {
+          "identifier": "INDEX",
+          "format": {
+            "type": "integer"
+          },
+          "hashing": {
+            "ngram": 1,
+            "weight": 0
+          }
+        },
+        {
+          "identifier": "NAME freetext",
+          "format": {
+            "type": "string",
+            "encoding": "utf-8",
+            "case": "mixed",
+            "minLength": 3
+          },
+          "hashing": {
+            "ngram": 2,
+            "weight": 0.5
+          }
+        },
+        {
+          "identifier": "DOB YYYY/MM/DD",
+          "format": {
+            "type": "string",
+            "encoding": "ascii",
+            "description": "Numbers separated by slashes, in the year, month, day order",
+            "pattern": "(?:\\d\\d\\d\\d/\\d\\d/\\d\\d)\\Z"
+          },
+          "hashing": {
+            "ngram": 1,
+            "positional": true
+          }
+        },
+        {
+          "identifier": "GENDER M or F",
+          "format": {
+            "type": "enum",
+            "values": ["M", "F"]
+          },
+          "hashing": {
+            "ngram": 1,
+            "weight": 2
+          }
+        }
+      ]
+    }
+
+
+Benchmark
+---------
+
+A quick hashing benchmark can be carried out to determine the rate at which the current machine
+can generate 10000 clks from a simple schema (data as generated :ref:`above <data-generation>`)::
+
+    python -m anonlinkclient.cli benchmark
```

Comment:
I think this can be `python -m anonlinkclient benchmark`? Or to be consistent with the above examples just `anonlink benchmark`.